src/sentry/event_manager.py

```diff
@@ -139,10 +140,26 @@ def has_pending_commit_resolution(group):
         .order_by("-datetime")
         .first()
     )
-    if recent_group_link is None:
+    if latest_issue_commit_resolution is None:
         return False
 
-    return not ReleaseCommit.objects.filter(commit__id=recent_group_link.linked_id).exists()
+    # commit has been released and is not in pending commit state
+    if ReleaseCommit.objects.filter(commit__id=latest_issue_commit_resolution.linked_id).exists():
+        return False
+    else:
+        # check if this commit is a part of a PR
+        pr_ids = list(
+            PullRequest.objects.filter(
+                pullrequestcommit__commit=latest_issue_commit_resolution.linked_id
+            ).values_list("id", flat=True)
+        )
+        if pr_ids:
+            # assume that this commit has been released if any commits in this PR have been released
+            if ReleaseCommit.objects.filter(
+                commit__pullrequestcommit__pull_request__in=pr_ids
```

Comment:
I think we can use `commit__pullrequestcommit__commit_id = latest_issue_commit_resolution.linked_id` here and skip the query for `pr_ids`. If there's a `PullRequestCommit` row then we know that the commit is associated with a `PullRequest`.

Reply:
@wedamija Since the intention is to retrieve all the commits in this PR that have been released, and `commit__pullrequestcommit__commit_id = latest_issue_commit_resolution.linked_id` would get the specific released commit associated with the PR, I’m not sure if skipping the query for `pr_ids` would work - I might be misunderstanding it though.

Reply:
Right, I forgot the specifics of what we're doing here. How about we
 - remove `if pr_ids:`. 
 - remove the `list` around the pull request query

This will cause the `pr_ids` queryset to be evaluated as part of the `ReleaseCommit` query, so it'll be roughly like
```
select rc.*
from release_commit rc
inner join commit c on c.id = rc.commit_id
inner join pullrequestcommit prc on prc.commit_id = c.id
where prc.pull_request_id in (
    select pr.id 
    from pullrequest pr
    inner join pullrequestcommit prc on prc.pull_request_id = pr.id
    where prc.commit_id = <commit_id>
)

```
Just saves us making an extra query here

Reply:
Just checking to make sure I understand how django magic is working. Please excuse the sloppy SQL that probably won't run properly, but hopefully it conveys the point.

This is what we have now with the two queries (slightly less efficient, but should work):
```
with pr_ids as (
	select pr.id from sentry_pullrequest pr
	inner join sentry_pullrequestcommit prc on pr.id=prc.pull_request_id
	inner join sentry_commit c on prc.commit_id = c.id and c.id=<linked_id>
)
select * from sentry_releasecommit rc
inner join sentry_commit c on rc.commit_id=c.id
inner join sentry_pullrequestcommit prc on prc.pull_request_id in (select * from pr_ids) 
```

Your suggestion:
```
select * from sentry_releasecommit rc
inner join sentry_commit c on rc.commit_id=c.id
inner join sentry_pullrequestcommit prc on prc.commit_id=c.id and prc.commit_id=<linked_id>
```

Is that how the SQL would compile down to for `commit__pullrequestcommit__commit_id`?

I think the distinction we wanna point out is the first one retrieves the PullRequest associated with the `linked_id` and expands out to check for any commits that were released within those PullRequests.

As opposed to what it looks like in your suggested query filter, which seems to check whether the `linked_id` is apart of a PullRequestCommit and is also a ReleaseCommit.

Reply:
Woah, so django would know to lazy eval the queries like that? Is that some pipelining going on, or it'll actually re-write the query? I.e. will it only reduce the number of round-trip calls or it'll actually result in a single query to postgres. 

That seems pretty cool.

Reply:
> Woah, so django would know to lazy eval the queries like that? Is that some pipelining going on, or it'll actually re-write the query? I.e. will it only reduce the number of round-trip calls or it'll actually result in a single query to postgres.
> 
> That seems pretty cool.

It doesn't do any pipelining that I know of, it just knows how to combine querysets together pretty well.

Django querysets are basically entirely lazy. So up until you do something that requires the results of the query they don't hit the database at all. So you can basically arbitrarily chain queries together and Django will just combine them into one single sql statement.